app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload # to get the statuses into the right order (since sorted :desc)
```

Comment:
Weird. If all we want if to get the statuses in the right order it should work. I suppose that `statuses(true)` also don't work right?